projects/canopy/src/lib/footer/footer.component.html

```diff
@@ -9,6 +9,7 @@ <h2 class="lg-visually-hidden" id="primary-links">Primary links</h2>
       <li *ngFor="let link of primaryLinks" class="lg-footer__primary-nav-item">
         <a
           (click)="handlePrimaryLinkClick($event)"
+          [attr.id]="link.id"
```

Comment:
should the id be optional in the `Link` ? it seems like it's needed otherwise it'll set the id to undefined

Reply:
Good point, it should be optional in this case, so I'll remove the attribute in the template if it's not provided

Reply:
Just checked and if no ID is provided it won't add the attribute, so it won't be undefined 👍 

We decided to make it optional as the links don't need a ID value, it's really up to the consiming app. Also making it required would be a breaking change.

Reply:
yeah, if you set an attribute to undefined, it won't be added, whereas `null` will